### DISCUSSION TITLE
Shared Variables vs. Source/Sink

### DISCUSSION THREAD

**Discussion by aentinger** [arduino-libraries/Arduino_Threads#88] [Polls]
If you are not familiar with those concepts and the way they are currently implemented in this library take a look at the [documentation](https://github.com/arduino-libraries/Arduino_Threads/blob/main/docs/02-data-exchange.md) and the [examples](https://github.com/arduino-libraries/Arduino_Threads/tree/main/examples/Threading_Basics).

* Which way of sharing data between threads would you prefer to use? Why?
* When using shared variables, was there any aspect that you would like to work differently? 
* When using source/sink, was there any aspect that you would like to work differently?

**Comment by dsyleixa** [comment]
I would prefer std::thread and std::mutex syntax because it's common and convenient standard syntax as stated here: https://github.com/arduino/language/discussions/2#discussioncomment-3897198
apart from mutexes for shared variables one might use std::atomic as well if suitable.
